src/main/java/knightminer/inspirations/common/datagen/CondRecipe.java

```diff
@@ -0,0 +1,314 @@
+		}
+
+		@Nonnull
+		@Override
+		public ResourceLocation getID() {
+			return id;
+		}
+
+		@Nonnull
+		@Override
+		public IRecipeSerializer<?> getSerializer() {
+			if (custSerial != null) {
+				return custSerial;
+			}
+			return recipe.getSerializer();
+		}
+
+		@Nullable
+		@Override
+		public JsonObject getAdvancementJson() {
+			return recipe.getAdvancementJson();
+		}
+
+		@Nullable
+		@Override
+		public ResourceLocation getAdvancementID() {
+			return recipe.getAdvancementID();
+		}
+	}
+
+	/**
+	 * Only applicable to shaped crafting, modifies it into a
+	 * retexturing recipe.
+	 */
+	private static class FinishedTexture extends Finished {
+		private final Ingredient texSource;
+		private final boolean matchFirst;
+
+		private FinishedTexture(ResourceLocation id, IFinishedRecipe recipe, boolean mirror, Ingredient texSource, boolean matchFirst, List<ICondition> cond) {
+			super(id, recipe, TextureRecipe.SERIALIZER, mirror, cond);
+			assert recipe.getSerializer() == ShapedRecipe.Serializer.CRAFTING_SHAPED;
+			this.texSource = texSource;
+			this.matchFirst = matchFirst;
+		}
+
+		@Nonnull
+		@Override
+		public JsonObject getRecipeJson() {
+			JsonObject json = super.getRecipeJson();
+			json.add("texture", texSource.serialize());
+			json.addProperty("match_first", matchFirst);
+			return json;
+		}
+	}
+
+	public static class ShapedBuilder extends ShapedRecipeBuilder {
+		private ArrayList<ICondition> conditions;
+		@Nullable
+		private IRecipeSerializer<?> custSerial;
+		@Nullable
+		private Ingredient textureSource;
+		private boolean textureMatchFirst;
+		private boolean mirror;
+
+		private ShapedBuilder(IItemProvider result, int count) {
+			super(result, count);
+			conditions = new ArrayList<>();
+			textureSource = null;
+			textureMatchFirst = false;
+			mirror = false;
+		}
+
+		/**
+		 * Vanilla option, whether the items can be horizontally mirrored.
+		 */
+		public ShapedBuilder canMirror() {
+			mirror = true;
+			return this;
+		}
+
+		public ShapedBuilder addCondition(ICondition cond) {
+			conditions.add(cond);
+			return this;
+		}
+
+		public ShapedBuilder custom(IRecipeSerializer<?> serializer) {
+			custSerial = serializer;
+			return this;
+		}
+
+		public ShapedBuilder textureSource(Ingredient ingredient) {
+			textureSource = ingredient;
+			return this;
+		}
+
+		public ShapedBuilder textureSource(IItemProvider item) {
+			return textureSource(Ingredient.fromItems(item));
+		}
+
+		public ShapedBuilder textureSource(Tag<Item> tag) {
+			return textureSource(Ingredient.fromTag(tag));
+		}
+
+		public ShapedBuilder textureMatchFirst() {
+			textureMatchFirst = true;
+			return this;
+		}
+
+		@Override
+		public void build(Consumer<IFinishedRecipe> consumer, ResourceLocation recipeLoc) {
+			// Capture the finished recipe which will be sent to the consumer.
+			final IFinishedRecipe[] output = {null};
+			super.build((res) -> output[0] = res, recipeLoc);
+
+			// It should have been called immediately.
```

Comment:
While this is currently true, wouldn't it be safer to just include the following code in the super consumer?

Reply:
That would be better, didn't think of that.